legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-memsql/src/main/java/org/finos/legend/engine/persistence/components/relational/memsql/sql/visitor/SchemaDefinitionVisitor.java

```diff
@@ -81,23 +85,48 @@ public VisitorResult visit(PhysicalPlanNode prev, SchemaDefinition current, Visi
             prev.push(constraint);
         }
 
-        for (Index idx : current.indexes())
+        // if table is sharded and primary keys are present
+        if (isTableColumnStore)
         {
-            TableConstraint constraint;
-            List<String> columns = idx.columns().stream().map(Field::name).collect(Collectors.toList());
-            if (idx.unique())
+            if (pkNum >= 1 && current.shardSpecification().isPresent() && current.shardSpecification().get().shardKeys().size() > 0)
```

Comment:
Can consider extracting this`current.shardSpecification().isPresent() && current.shardSpecification().get().shardKeys().size() > 0` into a variable called `isSharded` and reuse it at line 99

Reply:
fixed this